shared/rakefiles/secrets.rb

```diff
@@ -207,6 +227,65 @@ def self.fetch_secrets(encryption_key)
 
     return decrypted_secrets
   end
+
+  # This method creates new primary version for target encryption_key, and
+  # then disables all its previous versions.
+  def self.rotate_key(encryption_key)
+    puts "[secret-mgmt] Creating new primary version for key '#{encryption_key}'..."
+    new_version = %x{
+      gcloud kms keys versions create \
+      --location #{Secrets::KMS_LOCATION} \
+      --keyring #{Secrets::KMS_KEYRING} \
+      --key #{encryption_key} \
+      --primary --format json
+    }
+
+    begin
+      new_version = JSON.parse(new_version)
+      new_version_id = new_version["name"].match(/\/([0-9]+)$/)[1]
+    rescue
+      puts "ERROR: Unable to create new version for key '#{encryption_key}', terminating!"
```

Comment:
If it's safe to do so, it might be helpful to emit the JSON blob as part of this failure to assist debugging.

Reply:
Done.